testsuite/tests/name_resolution/iterated_component_association/iterassoc_filter.adb

```diff
@@ -0,0 +1,26 @@
+procedure Iterassoc_Filter is
+   type R is range 1 .. 5;
+   type A is array (R) of Float;
+
+   X : A := (1.0, 2.0, 3.0, 4.0, 5.0);
+
+   M : A := (for I in R when X (I) >= 3.0 => X (I) * X (I));
+   pragma Test_Statement;
+
+   O : A := (for F : Float of X when F >= 3.0 => F * F);
+   pragma Test_Statement;
+
+   P : A := (for F of X when F >= 3.0 => F * F);
+   pragma Test_Statement;
+
```

Comment:
Iterator filters are also valid in quantified expressions, could you add a little test case here? (It should work already with what you've done)

Reply:
As discussed live, those tests will be added later since they require additional development first.